src/Illuminate/Routing/Exceptions/UrlGenerationException.php

```diff
@@ -3,17 +3,35 @@
 namespace Illuminate\Routing\Exceptions;
 
 use Exception;
+use Illuminate\Routing\Route;
+use Illuminate\Support\Str;
 
 class UrlGenerationException extends Exception
 {
     /**
      * Create a new exception for missing route parameters.
      *
      * @param  \Illuminate\Routing\Route  $route
+     * @param array $parameters
      * @return static
      */
-    public static function forMissingParameters($route)
+    public static function forMissingParameters(Route $route, array $parameters = [])
     {
-        return new static("Missing required parameters for [Route: {$route->getName()}] [URI: {$route->uri()}].");
+        $paramterLabel = Str::plural('parameter', count($parameters));
+
+        $message = sprintf(
+            'Missing required %s for [Route: %s] [URI: %s]',
+            $paramterLabel,
+            $route->getName(),
+            $route->uri()
+        );
+
+        if (count($parameters) > 0) {
+            $message .= sprintf(' [Missing %s: %s]', $paramterLabel, implode(', ', $parameters));
+        }
```

Comment:
Small typo :)

```suggestion
        $parameterLabel = Str::plural('parameter', count($parameters));

        $message = sprintf(
            'Missing required %s for [Route: %s] [URI: %s]',
            $parameterLabel,
            $route->getName(),
            $route->uri()
        );

        if (count($parameters) > 0) {
            $message .= sprintf(' [Missing %s: %s]', $parameterLabel, implode(', ', $parameters));
        }
```

Reply:
Thank you so much. :) I had done a bunch of checks against this because I always miss that second `a`. :)